appinventor/components/src/com/google/appinventor/components/runtime/Ev3Motors.java

```diff
@@ -0,0 +1,664 @@
+                   category = ComponentCategory.LEGOMINDSTORMS,
+                   nonVisible = true,
+                   iconName = "images/legoMindstormsEv3.png")
+@SimpleObject
+public class Ev3Motors extends LegoMindstormsEv3Base {
+  private static final int DELAY_MILLISECONDS = 50;
+  private static final String DEFAULT_MOTOR_PORT = "ABC";
+  private static final double DEFAULT_WHEEL_DIAMETER = 4.32;
+
+  private int motorPortBitField = 1;
+  private double wheelDiameter = DEFAULT_WHEEL_DIAMETER;
+  private boolean regulationEnabled = true;
+  private boolean stopBeforeDisconnect = true;
+  private boolean tachoCountChangedEventEnabled = false;
+  private final Runnable sensorValueChecker;
+  private Handler eventHandler;
+  private int previousValue = 0;
+  private boolean ifReset = false;
+
+  /**
+   * Creates a new Ev3Motors component.
+   */
+  public Ev3Motors(ComponentContainer container) {
+    super(container, "Ev3Motors");
+    eventHandler = new Handler();
+    sensorValueChecker = new Runnable() {
+      public void run() {
+        String functionName = "";
+
+        if (bluetooth != null && bluetooth.IsConnected()) {
+          int sensorValue = getOutputCount(functionName, 0, motorPortBitField);
+
+          if (!ifReset) {
+            if (sensorValue != previousValue && tachoCountChangedEventEnabled) {
+              TachoCountChanged(sensorValue);
+            }
+          } else {
+            ifReset = false;
+          }
+          previousValue = sensorValue;
+        }
+
+        eventHandler.postDelayed(this, DELAY_MILLISECONDS);
+      }
+    };
+
+    eventHandler.post(sensorValueChecker);
+
+    MotorPort(DEFAULT_MOTOR_PORT);
+    StopBeforeDisconnect(true);
+    EnableSpeedRegulation(true);
+    WheelDiameter(DEFAULT_WHEEL_DIAMETER);
+    TachoCountChangedEventEnabled(false);
+  }
+
+  /**
+   * Returns the motor port.
+   */
+  @SimpleProperty(description = "The motor ports that the motors are connected to. The ports are specified by a sequence of port letters.",
+                  category = PropertyCategory.BEHAVIOR)
+  public String MotorPort() {
+    return bitFieldToMotorPortLetters(motorPortBitField);
+  }
+
+  /**
+   * Specifies the motor port.
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_STRING,
+                    defaultValue = DEFAULT_MOTOR_PORT)
+  @SimpleProperty
+  public void MotorPort(String motorPortLetters) {
+    String functionName = "MotorPort";
+    try {
+      motorPortBitField = motorPortLettersToBitField(motorPortLetters);
+    } catch (IllegalArgumentException e) {
+      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_MOTOR_PORT, motorPortLetters);
+    }
+  }
+
+  /**
+   * Specifies the diameter of the wheels attached on motors.
+   */
+  @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_FLOAT,
+                    defaultValue = "" + DEFAULT_WHEEL_DIAMETER)
```

Comment:
I think WheelDiameter should be userVisible = false, so it has to be set in the designer.